src/blop/agent.py

```diff
@@ -173,26 +173,20 @@ def train_inputs(self, index=None, **subset_kwargs):
 
     def raw_targets(self, index=None, **subset_kwargs):
         """
-        Get the raw, untransformed inputs for an objective (or for a subset).
+        Get the raw, untransformed targets for an objective (or for a subset of objectives).
         """
-        values = {}
-
-        for obj in self.objectives(**subset_kwargs):
-            # return torch.cat([self.raw_targets(index=obj.name) for obj in self.objectives(**subset_kwargs)], dim=-1)
-            values[obj.name] = torch.tensor(self._table.loc[:, obj.name].values, dtype=torch.double)
-
-        return values
+        if index is None:
+            return {obj.name: self.raw_targets(obj.name) for obj in self.objectives(**subset_kwargs)}
+        return torch.tensor(self._table.loc[:, self.objectives[index].name].values, dtype=torch.double)
 
-    def train_targets(self, concatenate=False, **subset_kwargs):
+    def train_targets(self, index=None, concatenate=False, **subset_kwargs):
```

Comment:
Same thing here, let's create a separate method that takes an index so as to not add complexity to the return type.